src/main/java/org/openrewrite/staticanalysis/ReplaceCollectWithStreamToList.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.staticanalysis;
+
+import java.time.Duration;
+import lombok.Value;
+import org.jetbrains.annotations.NotNull;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Option;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesJavaVersion;
+import org.openrewrite.java.search.UsesMethod;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.J.MethodInvocation;
+import org.openrewrite.java.tree.JRightPadded;
+
+@Value
+public class ReplaceCollectWithStreamToList extends Recipe {
+
+    private static final MethodMatcher STREAM_COLLECT_TO_LIST = new MethodMatcher("java.util.stream.Stream collect(java.util.stream.Collector)");
+    private static final MethodMatcher COLLECTORS_TO_LIST = new MethodMatcher("java.util.stream.Collectors toList()");
+    private static final MethodMatcher COLLECTORS_TO_UNMODIFIABLE_LIST = new MethodMatcher("java.util.stream.Collectors toUnmodifiableList()");
+
+    @Option(displayName = "Should the recipe also apply to mutable toList?",
+        description = "Also replace Java 11 `Stream.collect(Collectors.toList())` with Java 16 `Stream.toList()`.",
+        required = false)
+    @Nullable
+    Boolean includeMutable;
+
+    @Override
+    public String getDisplayName() {
+        return "Replace Stream.collect(Collectors.toUnmodifiableList()) with Stream.toList()";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Replace Java 11 `Stream.collect(Collectors.toUnmodifiableList())` with Java 16 `Stream.toList()`.";
+    }
+
+    @Override
+    public Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(1);
+    }
+
```

Comment:
Would be nice to link to the RSPEC here as well
```suggestion

    @Override
    public Set<String> getTags() {
        return new HashSet<>(Arrays.asList("RSPEC-6204"));
    }

```